File: JULY-2025/19-07-2025/p1233.kt
class Solution {
    fun removeSubfolders(folder: Array<String>): List<String> {
        folder.sort()
        val res = mutableListOf<String>()
        res.add(folder[0])
        var prevSlash = folder[0] + '/'

        for (i in 1..<folder.size) {
            if (!folder[i].startsWith(prevSlash)) {
                res.add(folder[i])
                prevSlash = folder[i] + '/'
            }
        }
        return res
    }
}

File: JULY-2025/19-07-2025/p1233.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public IList<string> RemoveSubfolders(string[] folder)
    {
        Array.Sort(folder, StringComparer.Ordinal);
        var res  = new List<string>(folder.Length);
        string prev = string.Empty;

        foreach (string path in folder)
        {
            if (prev.Length == 0 ||
                !path.StartsWith(prev, StringComparison.Ordinal) ||
                path.Length == prev.Length ||
                path[prev.Length] != '/')
            {
                res.Add(path);
                prev = path;
            }
        }

        return res;
    }
}